sdcm/nemesis.py

```diff
@@ -4126,6 +4127,44 @@ def disrupt_create_index(self):
             finally:
                 drop_index(session, ks, index_name)
 
+    def disrupt_add_remove_mv(self):
+        """
+        Create a Materialized view on an existing table while a node is down.
+        Take node up and run a repair.
+        Verify the MV can be used in a query.
+        Finally, drop the MV.
+        """
+        node1, node2 = self.cluster.nodes[:2]
+        self.log.info("Stopping Scylla on node1")
+        ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=node2, filter_empty_tables=True, filter_out_mv=True)
+        if not ks_cfs:
+            raise UnsupportedNemesis(
+                'Non-system keyspace and table are not found. nemesis can\'t be run')
+        ks_name, base_table_name = random.choice(ks_cfs).split('.')
+        view_name = f'{base_table_name}_view'
+        node1.stop_scylla()
+        with self.cluster.cql_connection_patient(node2) as session:
+            partition_key_name = get_partition_key_name(session=session, ks=ks_name, cf=base_table_name)
```

Comment:
what in case of compound partition key? why not to take whole list instead of single column name as str and pass it to `create_materialized_view`?

Reply:
@soyacz , i'm not sure what you mean exactly, but this is how a materialized view can be basically applied, so i wanted to test the common case that would work for all user tables. The main challenge we concern here, is the dynamic building of a new view for and existing table under stress.

Reply:
Ah, perhaps you meant according to the code there's a returned list it chooses from, but i think we'll always expect it to have a single value.